msg/trajectory_waypoint.msg

```diff
@@ -0,0 +1,38 @@
+# Waypoints Trajectory description. See also Mavlink TRAJECTORY msg
+# The topic trajectory_waypoint_desired is used to send the user desired waypoints from the position controller to the companion computer / avoidance module.
+# The topic trajectory_waypoint is used to send the adjusted waypoints from the companion computer / avoidance module to the position controller.
+
+uint8 POINT_0 = 0
+uint8 POINT_1 = 1
+uint8 POINT_2 = 2
+uint8 POINT_3 = 3
+uint8 POINT_4 = 4
+
+uint8 X = 0
```

Comment:
At the moment these become DEFINES if included in C. We should probably change that given the generic naming.

Reply:
@dagar  I see. The idea behind those was that indexes are quite cryptic to read. What about having separate methods to handle position, velocity, acceleration, and yaw such as https://github.com/mrivi/Firmware/commit/8c699b6522ed3334fc1a3dceca07bbf2b9ca8954#diff-fc77c3ef569029d45764664c75d4b0c1R3306 .
So there is the meaningless index bu the name of the method helps reading the code. What do you think?

Reply:
I need to go over the entire PR to build a better mental model, but one idea to consider is adding a bit more structure. 

For example these generated messages can be composed of other generated messages. You could have a trajectory waypoint definition message, and then the main message for uORB publication would contain an array of them.


	trajectory_waypoint
	  - float32 x, y, z (or an array: float32[3] position)
	  - float32 vx, vy, vz
	  - float32 ax, ay, az
	  - float32 yaw
	  - float32 yawspeed

Then have a different message (vehicle_trajectory_waypoints or something) containing them.

	vehicle_trajectory_waypoints
	  - px4/trajectory_waypoint[11] waypoints


Using an array of floats for position, velocity, acceleration can be convenient if directly creating Vectors.

``` c++
matrix::Vector3f{trajectory_waypoint.x, trajectory_waypoint.y, trajectory_waypoint.z}
// becomes
matrix::Vector3f{trajectory_waypoint.position}
```

Reply:
The problem comes from when using it in C. ( This is the relevant line I think https://github.com/PX4/Firmware/blob/ab279d5fe9e014cec26add8a56bd5e676152b1c5/msg/templates/uorb/msg.h.template#L86 ) Maybe we should add a prefix for constants in C with the message name directly in the template.

Reply:
@thomasgubler yes that was my initial thought. It should be a quick/easy change because we're hardly using them at all in PX4 (in C). That makes me question dropping the C definition entirely, but I won't go there right now...

Reply:
https://github.com/PX4/Firmware/pull/9641